### PR TITLE
groups

### PR DESCRIPTION
**Pull request by harelglik**

### CODE REVIEW

File: packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.test.ts [hunk 1]
```diff
@@ -41,56 +37,46 @@ const ALL_APPOINTMENT_IDS = [
...
 beforeAll(reset);
 afterEach(reset);
-
 test("Unauthenticated user throws exception", async () => {
   const action = () => callFunction();
   await expectAsyncThrows(action, "Unauthorized");
 });
 
-test("User that is not admin throws exception", async () => {
+test("User that is not coordinator throws exception", async () => {
```
Comment (yaron1m) [inline packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.test.ts:53]: Should you add more tests to this file? GetCoordinatorAppointmentsHandler changed a lot

File: packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts [hunk 2]
```diff
@@ -28,11 +31,69 @@ export default async function (
...
+    case CoordinatorRole.GROUP_COORDINATOR:
+      break;
+  }
+}
+
+async function filterDonorsInGroup(
+  coordinator: DbCoordinator,
+  donorsInAppointments: DbDonor[]
+) {
+  const groupIds = await GroupDAL.getGroupIdsOfCoordinatorId(coordinator.id);
```
Comment (yaron1m) [inline packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts:85]: Should this be Set?

File: packages/functions/src/coordinator/GetDonorsHandler.ts [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
+  const coordinator = await CoordinatorDAL.getCoordinator(callerId);
   if (!coordinator) {
     console.error("Could not find calling user", callerId);
     throw Error(`User ${callerId} is not an admin`);
   }
 
-  const donors = await getAllDonors();
+  let donors: DbDonor[] = [];
+
+  switch (coordinator.role) {
```
Comment (yaron1m) [inline packages/functions/src/coordinator/GetDonorsHandler.ts:22]: Tests?

File: packages/functions/src/react-app-env.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (yaron1m) [inline packages/functions/src/react-app-env.d.ts:1]: Remove this file

File: packages/functions/tsconfig.json [hunk 5]
```diff
@@ -1,3 +1,24 @@
 {
-  "extends": "../../tsconfig.json"
+  "extends": "../../tsconfig.json",
```
Comment (yaron1m) [inline packages/functions/tsconfig.json:2]: Revert this change

File: packages/functions/src/react-app-env.d.ts (hunk 4 above)
Comment (harelglik) [reply packages/functions/src/react-app-env.d.ts:1]: Done

File: packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.test.ts (hunk 1 above)
Comment (harelglik) [reply packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.test.ts:67]: Sorry did not merge the test files correctly, there is now a test that checks donors in groups

File: packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts (hunk 2 above)
Comment (harelglik) [reply packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts:85]: Done
There are many places where we should use sets but use arrays.

**Review by yaron1m — APPROVED** [review]
